numpy/core/memmap.py

```diff
@@ -322,10 +326,72 @@ def flush(self):
         See Also
         --------
         memmap
+        """
+        if self._mmap is None:
+            raise TypeError(
+                    "flush cannot be used, because the memory map it "
+                    "is not backed by any memory.")
```

Comment:
The error seems a bit confusing.  Maybe just "not actually backed by a memory map"?